### PR TITLE
 feat: UI changes for Apicurio integration

### PR DESCRIPTION
**Pull request by hemahg**
feat: UI changes for Apicurio integration

### CODE REVIEW

File: ui/components/MessagesTable/MessagesTable.tsx [hunk 1]
```diff
@@ -264,30 +268,42 @@ export function MessagesTable({
...
+                                />
+                                {row.relationships.keySchema && (
+                                  <TextContent>
+                                    <Text component={"small"}>
+                                      {row.relationships.keySchema?.meta
+                                        ?.name && (
+                                        <ExternalLink
+                                          testId={"key-schema"}
+                                          href={`${baseurl}/${row.relationships.keySchema?.links?.content}`}
+                                        >
```
Comment (MikeEdgar) [inline ui/components/MessagesTable/MessagesTable.tsx:287]: It seems a little funky to nest the schema under the topic messages, but I realize that there aren't really any alternatives given what the API is returning for the link. Maybe in a future PR the API can return the registry name and schema ID as separate fields and the pages no longer need to be nested this way.

File: ui/api/schema/action.ts [hunk 2]
```diff
@@ -0,0 +1,12 @@
+"use server";
+
+import { getHeaders } from "../api";
+
+export async function getSchema(contentLink: string) {
+  const url = `${process.env.BACKEND_URL}/${contentLink}`;
+  const res = await fetch(url, {
+    headers: await getHeaders(),
+  });
+  const rawData = await res.json();
```
[the remark this answers is not in the document]
Comment (hemahg) [reply ui/api/schema/action.ts]: fixed

File: ui/app/[locale]/(authorized)/kafka/[kafkaId]/topics/[topicId]/messages/[...content]/page.tsx [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+}: {
+  params: { content: string[] };
+}) {
+  const fullPath = params.content.join("/");
+  const content = await getSchema(fullPath);
+  const { namespace, name } = content;
+  const schemaName = `${namespace}.${name}`;
+  return (
+    <ConnectedSchema
+      content={JSON.stringify(content, null, 2)}
```
[the remark this answers is not in the document]
Comment (hemahg) [reply ui/app/[locale]/(authorized)/kafka/[kafkaId]/topics/[topicId]/messages/[...content]/page.tsx]: fixed

File: ui/components/MessagesTable/components/MessageDetails.tsx [hunk 4]
```diff
@@ -143,6 +183,18 @@ export function MessageDetailsBody({
               )}
             </DescriptionListDescription>
           </DescriptionListGroup>
+          <DescriptionListGroup>
+            <DescriptionListTerm>{t("field.key-format")}</DescriptionListTerm>
+            <DescriptionListDescription>
+              {message.relationships.keySchema?.meta?.artifactType}
```
[the remark this answers is not in the document]
Comment (hemahg) [reply ui/components/MessagesTable/components/MessageDetails.tsx]: updated

File: ui/components/MessagesTable/components/MessageDetails.tsx [hunk 5]
```diff
@@ -143,6 +183,18 @@ export function MessageDetailsBody({
...
+            </DescriptionListDescription>
+          </DescriptionListGroup>
+          <DescriptionListGroup>
+            <DescriptionListTerm>{t("field.value-format")}</DescriptionListTerm>
+            <DescriptionListDescription>
+              {message.relationships.valueSchema?.meta?.artifactType}
```
[the remark this answers is not in the document]
Comment (hemahg) [reply ui/components/MessagesTable/components/MessageDetails.tsx]: Done

File: ui/components/MessagesTable/MessagesTable.tsx [hunk 6]
```diff
@@ -264,30 +272,49 @@ export function MessagesTable({
...
+                                {row.relationships.keySchema && (
+                                  <TextContent>
+                                    <Text component={"small"}>
+                                      {row.relationships.keySchema?.meta
+                                        ?.name &&
+                                      row.relationships.keySchema?.links
+                                        ?.content ? (
+                                        <ExternalLink
+                                          testId={"key-schema"}
+                                          href={`${baseurl}/schema/content=${encodeURIComponent(row.relationships.keySchema?.links?.content)}&schemaname=${encodeURIComponent(
```
Comment (MikeEdgar) [inline ui/components/MessagesTable/MessagesTable.tsx:292]:
```suggestion
                                          href={`${baseurl}/schema?content=${encodeURIComponent(row.relationships.keySchema?.links?.content)}&schemaname=${encodeURIComponent(
```

**Review by MikeEdgar — CHANGES_REQUESTED** [review]
Very nice, there is just one typo that I noticed. Otherwise I think we can merge. Thanks

File: ui/components/MessagesTable/MessagesTable.tsx (hunk 6 above)
Comment (hemahg) [reply ui/components/MessagesTable/MessagesTable.tsx]: sorry, fixed it

**Review by MikeEdgar — APPROVED** [review]
:+1:

**Pull request merged**